src/ssl.c

```diff
@@ -15797,6 +15900,21 @@ int wolfSSL_set_compression(WOLFSSL* ssl)
     }
 
 
+    int wolfSSL_set0_verify_cert_store(WOLFSSL *ssl, WOLFSSL_X509_STORE* str)
+    {
+        WOLFSSL_ENTER("wolfSSL_set0_verify_cert_store");
+
+        if (ssl == NULL || ssl->ctx == NULL || str == NULL) {
+            WOLFSSL_MSG("Bad parameter");
+            return WOLFSSL_FAILURE;
+        }
+        WOLFSSL_MSG("WARNING: This function modifies the cert store used by all "
```

Comment:
This looks dangerous. Openresty uses the same store for all SSL objects?

Reply:
It probably is. I'm guessing its better to guard it with some kind of an ALLOW_UNSAFE macro.

Reply:
Todd and I agree that adding a pointer to the store in the SSL would be fine.
Please do this.

Reply:
How should I change the existing references to `ssl->ctx->cm` and others to use the WOLFSSL store? I was thinking of defining macros like:
```
#define SSL_GET_STORE(ssl) (ssl->x509_store_pt ? ssl->x509_store_pt : ssl->ctx->x509_store_pt)
#define SSL_GET_CM(ssl) (ssl->x509_store_pt ? ssl->x509_store_pt->cm : ssl->ctx->cm)
```
What do you think of this idea?

Reply:
The pointer to the store (cm) will be copied into the WOLFSSL object on wolfSSL_new().
If you need to modify the store locally, then duplicate it and mark for free.
Make sense?

Reply:
I'm not sure if copying the cm is a good idea. This would break existing connections if the CTX store is set to a new value. If we would up the refCount for every SSL object then a new cm in the CTX would still not be registered in the SSL objects. If you think that this is acceptable behaviour then I will go ahead and change it.

Reply:
Sorry. Forgot what the API did.
Yeah just replace the pointer.

Reply:
So using the macros I suggested earlier? I'm sorry I don't understand what you mean by "just replace the pointer".

Reply:
Why macro?
Copy the pointer down in the InitSSL.

Reply:
Ok. And just to clarify, it is ok if a new cm is set in the ctx and this is not registered by existing ssl objects? I'm asking because this changes how the cm functions currently.

Reply:
After talking to Todd, he would be happy with up reffing (incrementing reference count) on the CertManager.
The WOLFSSL_X509_STORE may need to have have a reference count as well.
Use a  macro as you have above. That is, if the SSL object doesn't have a store then use the context's store.
If the store being set is the one in the context then don't do anything.

Reply:
Thanks for the clarification Sean! This clears everything up.

Reply:
In the last commit (a5a5b3d4678a3e3408132066118ca02305aefd65) I added a `WOLFSSL_X509_STORE` to the SSL object, added a ref count, and substituted calls with macros choosing the appropriate object.